### ISSUE TITLE
Cannot map array to Json dictionary

### ISSUE THREAD

**Issue by Taco55** [Hearst-DD/ObjectMapper#568]
I have an issue with ObjectMapper to map an array to a Json dictionary (i.e. for use with Alamofire).
However, the output I get is a `SwiftDeferredNSArray` which I cannot explain. I already tried a custom transform but it did not help.

When the array is mapped to a Json string the output is correct.

What would be the proper way to map an array to a Json dictionary?

```
class Example: Mappable {
    var foo: String = "bar"
    var qux = [0,1,2]

    required convenience init?(_ map: Map) {
        self.init()
    }
    
    func mapping(map: Map) {
        self.foo <- map["foo"]
        self.qux <- map["qux"]
    }
}


class ViewController: UIViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let example = Example()
        
        print( Mapper().toJSON(example) )
        /* 
        (Incorrect) output:
        ["qux": <_TtCs21_SwiftDeferredNSArray 0x7fcf5be42d30>(
            0,
            1,
            2
            )
            , "foo": bar]
        */

        
       print( Mapper().toJSONString(example, prettyPrint: true)! )
       /* (
       Correct) output:
       {
           "qux" : [
               0,
               1,
               2
            ],
            "foo" : "bar"
        }
        */
    }  
}
```

**Comment by Taco55** [comment]
I just found that a `SwiftDeferredNSArray` results when the values in a Swift dictionary are defined with `AnyObject`. However, when `Any` is used, the output is correct.

```
var fields = [String: AnyObject]()
fields["key1"] = [1,2,3,4]
print(fields) // Output: ["key": <_TtCs21_SwiftDeferredNSArray 0x7fd9db516940>(\n1,\n2,\n3,\n4\n)\n]

var fields = [String: Any]()
fields["key1"] = [1,2,3,4]
print(fields) // Output ["key": [1, 2, 3, 4]]
```

Any suggestions how to handle this with ObjectMapper?

**Comment by tobiasoleary** [comment]
Does the SwiftDefferedArrray Type implement mappable?

**Comment by tobiasoleary** [comment]
Nevermind. Being dumb.

**Comment by Taco55** [comment]
Never mind. Thanks for the reply.

With Swift 3 the value type in the Swift dictionary is changed from AnyObject to Any and does not result in `SwiftDeferredNSArray`s anymore. Because my problem is solved after conversion to Swift 3.0 and new version of ObjectMapper I will close this issue.
